File: iOS/ssabree/ssabree/Core/Repository/CommentRepository.swift
import Foundation

protocol CommentRepository {
    func like(commentId: Int) async -> Result<CommentLikeModel, Error>
    func unlike(commentId: Int) async -> Result<CommentLikeModel, Error>
    func updateComment(commentId: Int, content: String) async -> Result<Void, Error>
    func deleteComment(commentId: Int) async -> Result<Void, Error>
}

File: android/ssabree/app/src/main/java/com/ssafy/ssabree/core/repository/CommentRepository.kt
package com.ssafy.ssabree.core.repository

import com.ssafy.ssabree.core.repository.model.CommentLikeModel

interface CommentRepository {
    suspend fun updateComment(commentId: Long, content: String): Result<Unit>
    suspend fun deleteComment(commentId: Long): Result<Unit>
    suspend fun like(commentId: Long): Result<CommentLikeModel>
    suspend fun unlike(commentId: Long): Result<CommentLikeModel>
}
